File: TelnetServer.java
import java.io.*;
import java.net.*;
import java.util.Scanner;

public class TelnetServer {
    public static void main(String[] args) {
        try (ServerSocket serverSocket = new ServerSocket(2323)) {
            System.out.println("Telnet Server started on port 2323...");
            Socket socket = serverSocket.accept();
            System.out.println("Client connected...");

            BufferedReader reader = new BufferedReader(new InputStreamReader(socket.getInputStream()));
            PrintWriter writer = new PrintWriter(socket.getOutputStream(), true);
            
            writer.println("Welcome to the Telnet Server. Type your message:");
            String receivedData;
            while ((receivedData = reader.readLine()) != null) {
                System.out.println("Received: " + receivedData);
                writer.println("Echo: " + receivedData);
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: backend_js/telnetServer.js
import net from "net"

const server = net.createServer((socket) => {
  console.log("Client connected");

  socket.write("Welcome to the Telnet Server!\n");

  socket.on("data", (data) => {
    console.log("Received:", data.toString());
    socket.write("Echo: " + data);
  });

  socket.on("end", () => {
    console.log("Client disconnected");
  });
});

server.listen(2323, () => {
  console.log("Telnet Server running on port 2323...");
});
